qase-python-commons/src/qase/commons/profilers/network.py

```diff
@@ -68,15 +70,25 @@ def _log_pre_request(request):
             data=StepRequestData(
                 request_method=request.method,
                 request_url=request.url,
-                request_body=request.body,
-                request_headers=request.headers,
+                request_body=request.body if hasattr(request, 'body') else None,
+                request_headers=request.headers if hasattr(request, 'headers') else None
             ),
         )
 
     @staticmethod
     def _log_post_response(response, url=None, *args, **kwargs):
         status = response.status if hasattr(response, 'status') else response.status_code
         profiler = NetworkProfilerSingleton.get_instance()
+        if profiler.step is None:
+            profiler.step = Step(
+                id=str(uuid.uuid4()),
+                step_type=StepType.REQUEST,
+                data=StepRequestData(
```

Comment:
Will request and response be reported as two adjacent steps or a single step?

Reply:
in a single step